integration_tests/abstract_account_test.go

```diff
@@ -0,0 +1,214 @@
+	"github.com/stretchr/testify/require"
+
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	aatypes "github.com/larry0x/abstract-account/x/abstractaccount/types"
+)
+
+func TestXionAbstractAccount(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+
+	t.Parallel()
+	td := BuildXionChain(t, "0.0uxion", ModifyInterChainGenesis(ModifyInterChainGenesisFn{ModifyGenesisShortProposals}, [][]string{{votingPeriod, maxDepositPeriod}}))
+	xion, ctx := td.xionChain, td.ctx
+
+	// Register All messages we are interacting.
+	xion.Config().EncodingConfig.InterfaceRegistry.RegisterImplementations(
+		(*types.Msg)(nil),
+		&xiontypes.MsgSetPlatformPercentage{},
+		&xiontypes.MsgSend{},
+		&wasmtypes.MsgInstantiateContract{},
+		&wasmtypes.MsgExecuteContract{},
+		&wasmtypes.MsgStoreCode{},
+		&aatypes.MsgUpdateParams{},
+		&aatypes.MsgRegisterAccount{},
+	)
+
+	xion.Config().EncodingConfig.InterfaceRegistry.RegisterImplementations((*authtypes.AccountI)(nil), &aatypes.AbstractAccount{})
+	xion.Config().EncodingConfig.InterfaceRegistry.RegisterImplementations((*cryptotypes.PubKey)(nil), &aatypes.NilPubKey{})
+
+	// Create and Fund User Wallets
+	t.Log("creating and funding user accounts")
+	fundAmount := int64(10_000_000)
+	users := ibctest.GetAndFundTestUsers(t, ctx, "default", fundAmount, xion)
+	xionUser := users[0]
+	currentHeight, _ := xion.Height(ctx)
+	testutil.WaitForBlocks(ctx, int(currentHeight)+8, xion)
+	t.Logf("created xion user %s", xionUser.FormattedAddress())
+
+	xionUserBalInitial, err := xion.GetBalance(ctx, xionUser.FormattedAddress(), xion.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, fundAmount, xionUserBalInitial)
+
+	// Create a Secondary Key For Rotation
+	recipientKeyName := "recipient-key"
+	err = xion.CreateKey(ctx, recipientKeyName)
+	require.NoError(t, err)
+	receipientKeyAddressBytes, err := xion.GetAddress(ctx, recipientKeyName)
+	require.NoError(t, err)
+	recipientKeyAddress, err := types.Bech32ifyAddressBytes(xion.Config().Bech32Prefix, receipientKeyAddressBytes)
+	require.NoError(t, err)
+
+	// Get Public Key For Funded Account
+	account, err := ExecBin(t, ctx, xion.FullNodes[0],
+		xionUser.KeyName(),
+		"keys", "show",
+		xionUser.KeyName(),
+		"--keyring-backend", keyring.BackendTest,
+		"-p",
+	)
+	require.NoError(t, err)
+
+	// Store Wasm Contract
+	codeID, err := xion.StoreContract(ctx, xionUser.FormattedAddress(), "./testdata/contracts/account_updatable-aarch64.wasm")
+	require.NoError(t, err)
+
+	depositedFunds := fmt.Sprintf("%d%s", 100000, xion.Config().Denom)
+
+	// Register Abstract Account Using Public Key
+	registeredTxHash, err := ExecTx(t, ctx, xion.FullNodes[0],
+		xionUser.KeyName(),
+		"abstract-account", "register",
+		codeID,
+		fmt.Sprintf(`{"pubkey": "%s"}`, account["key"]),
```

Comment:
The definition for the init msg is 
```
#[cw_serde]
pub struct InstantiateMsg {
    pub id: u8,
    pub authenticator: Authenticator,
    pub signature: Binary,
}
```

not sure why this isnt panicking on deserialization

Reply:
Different contract. Will refactor to use this contract and msg.